mne/time_frequency/multitaper.py

```diff
@@ -543,12 +542,68 @@ def multitaper_psd(x, sfreq=2 * np.pi, fmin=0, fmax=np.inf, bandwidth=None,
                        for x in np.array_split(x_mt, n_jobs))
         psd = np.concatenate(out)
 
-    if x.ndim == 1:
-        # return a 1d array if input was 1d
-        psd = psd[0, :]
-
     freqs = freqs[freq_mask]
     if normalization == 'full':
         psd /= sfreq
 
+    # Combining/reshaping to original data shape
+    psd = psd.reshape(np.hstack([dshape, -1]))
+    if ndim_in == 1:
+        psd = psd.squeeze()
     return psd, freqs
+
+
+@deprecated('This will be deprecated in release v0.13, see psd_multitaper.')
+@verbose
+def multitaper_psd(x, sfreq=2 * np.pi, fmin=0, fmax=np.inf, bandwidth=None,
+                   adaptive=False, low_bias=True, n_jobs=1,
+                   normalization='length', verbose=None):
+    """Compute power spectrum density (PSD) using a multi-taper method
+
+    Parameters
+    ----------
+    x : array, shape=(n_signals, n_times) or (n_times,)
+        The data to compute PSD from.
+    sfreq : float
+        The sampling frequency.
+    fmin : float
+        The lower frequency of interest.
+    fmax : float
+        The upper frequency of interest.
+    bandwidth : float
+        The bandwidth of the multi taper windowing function in Hz.
+    adaptive : bool
+        Use adaptive weights to combine the tapered spectra into PSD
+        (slow, use n_jobs >> 1 to speed up computation).
+    low_bias : bool
+        Only use tapers with more than 90% spectral concentration within
+        bandwidth.
+    n_jobs : int
+        Number of parallel jobs to use (only used if adaptive=True).
+    normalization : str
+        Either "full" or "length" (default). If "full", the PSD will
+        be normalized by the sampling rate as well as the length of
+        the signal (as in nitime).
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    psd : array, shape=(n_signals, len(freqs)) or (len(freqs),)
+        The computed PSD.
+    freqs : array
+        The frequency points in Hz of the PSD.
+
+    See Also
+    --------
+    mne.io.Raw.plot_psd, mne.Epochs.plot_psd
+
+    Notes
+    -----
+    .. versionadded:: 0.9.0
```

Comment:
maybe the 1993 one ... our code was copied from nitime, so you may want to
look into their references.

On Wed, Dec 30, 2015 at 12:50 AM, Chris Holdgraf <notifications@github.com>
wrote:

> In mne/time_frequency/multitaper.py
> <https://github.com/mne-tools/mne-python/pull/2710#discussion_r48563028>:
>
> > +        If not None, override default verbose level (see mne.verbose).
> > +
> > +    Returns
> > +    -------
> > +    psd : array, shape=(n_signals, len(freqs)) or (len(freqs),)
> > +        The computed PSD.
> > +    freqs : array
> > +        The frequency points in Hz of the PSD.
> > +
> > +    See Also
> > +    --------
> > +    mne.io.Raw.plot_psd, mne.Epochs.plot_psd
> > +
> > +    Notes
> > +    -----
> > +    .. versionadded:: 0.9.0
>
> we could do either reference 2 or 3 from here:
> http://www.fieldtriptoolbox.org/tutorial/timefrequencyanalysis#fn__3
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/2710/files#r48563028>.
>